genai-perf/genai_perf/TCO_calculator/TCO_calculator.ipynb

```diff
@@ -0,0 +1,621 @@
+    "}\n",
+    "\n",
+    "# Iterate over all defined use cases and run the benchmark script for each\n",
+    "for description in \"${!useCases[@]}\"; do\n",
+    "    runBenchmark \"$description\"\n",
+    "done\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "603f1941-5206-4bca-a547-028e0ea50f21",
+   "metadata": {},
+   "source": [
+    "Next, we execute the bash script, which will carry out the defined benchmarking scenarios and gather the data in a default directory named `artifacts` under the current working directory."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6cbfacd3-5755-4c0b-ae23-3abffceebbdb",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "bash benchmark.sh"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c480b28d-6816-4c84-9124-bdc56fc81f41",
+   "metadata": {},
+   "source": [
+    "## Reading gen-AI-perf data\n",
+    "\n",
+    "Once performance benchmarking is done, we read and collect the results in a single data frame."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "3815b7f1-b51b-44ea-bbbc-b023ec0beeca",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gen_AI_perf_field = [\n",
+    " 'Inter Token 90th Percentile Latency (ms)',\n",
+    " 'Inter Token 99th Percentile Latency (ms)',\n",
+    " 'Inter Token Average Latency (ms)',\n",
+    " 'Time to First Token 90th Percentile Latency (ms)',\n",
+    " 'Time to First Token 99th Percentile Latency (ms)',\n",
+    " 'Time to First Token Average Latency (ms)',\n",
+    " 'Request 90th Percentile Latency (ms)',\n",
+    " 'Request 99th Percentile Latency (ms)',\n",
+    " 'Request Latency (ms)',\n",
+    " 'Requests per Second',\n",
+    " 'Tokens per Second']\n",
+    "\n",
+    "# Other experimental params: 'Seq Length (ISL/OSL)', 'Concurrency',"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "ff69c986-0c9b-46a9-8f28-800cd61ab24d",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/tmp/ipykernel_203/961144172.py:38: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.\n",
+      "  df = pd.concat([df, pd.DataFrame([row])], ignore_index=True)\n"
+     ]
+    }
+   ],
+   "source": [
+    "import os\n",
+    "import json\n",
+    "import pandas as pd\n",
+    "\n",
+    "root_dir = \"./artifacts\"\n",
+    "directory_prefix = \"meta_llama-3.1-8b-instruct-openai-chat-concurrency\" # Change this to fit the actual model deployed\n",
+    "\n",
+    "ISL_OSL_list = [\"200_5\", \"200_200\", \"1000_200\", \"200_1000\"]\n",
+    "concurrencies = [1, 2, 5, 10, 50, 100, 250]\n",
```

Comment:
consider capitalizing `CONCURRENCIES` to match ISL_OSL

Reply:
agreed